src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -982,7 +983,7 @@ private List<BinaryExpressionNode> BuildNullAssignments(List<AssociativeNode> as
                 }
 
                 IdentifierNode leftNode = bNode.LeftNode as IdentifierNode;
-                if (leftNode == null || leftNode.ArrayDimensions != null)
+                if (leftNode == null || leftNode.ArrayDimensions != null  || bNode.Optr != Operator.assign)
```

Comment:
Minor, but why not have this check up here: https://github.com/DynamoDS/Dynamo/pull/10512/files#diff-2639b9fb615a6392f1215b5d9ff3c0f2R980

Reply:
you're right, that seems like a more appropriate place, where the binary node is validated, I can move it.